tests/grid/get_coarse_mesh_description_01.cc

```diff
@@ -0,0 +1,109 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// Test GridTools::get_coarse_mesh_description()
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+
+#include "../tests.h"
+
+
+
+template <int dim>
+void
+test()
+{
+  {
+    // Check that we can copy a coarse Triangulation
+    Triangulation<dim> tria;
+    GridGenerator::hyper_ball(tria);
+    // Make the boundary id check more robust
+    for (auto face : tria.active_face_iterators())
+      if (face->at_boundary() && 0.0 < face->center()[0])
+        face->set_boundary_id(42);
+
+    std::vector<Point<dim>>    vertices;
+    std::vector<CellData<dim>> cells;
+    SubCellData                subcell_data;
+    std::tie(vertices, cells, subcell_data) =
+      GridTools::get_coarse_mesh_description(tria);
+
+    Triangulation<dim> tria_2;
+    tria_2.create_triangulation(vertices, cells, subcell_data);
+    Assert(GridTools::have_same_coarse_mesh(tria, tria_2), ExcInternalError());
+
+    GridOut grid_out;
+    deallog << "Original Triangulation:" << std::endl;
+    grid_out.write_vtk(tria, deallog.get_file_stream());
+    deallog << "Triangulation constructed from an unrefined Triangulation:"
+            << std::endl;
+    grid_out.write_vtk(tria_2, deallog.get_file_stream());
```

Comment:
you could also borrow the `verify()` function from the `tests/serialization/serialization.h` file to check that the two triangulations are the same.

Reply:
I think I'll stick with VTK for now: its another test on top of the recent rewrite. I do assert that the coarse grids are equal too.